src/main/java/me/moodcat/database/BulkInsertData.java

```diff
@@ -0,0 +1,191 @@
+
+    /**
+     * The song DAO provider.
+     */
+    private final Provider<SongDAO> songDAOProvider;
+
+    /**
+     * The room DAO provider.
+     */
+    private final Provider<RoomDAO> roomDAOProvider;
+
+    /**
+     * The SoundCloud extraction API.
+     */
+    private final SoundCloudExtract soundCloudExtract;
+
+    /**
+     * Object to bulk insert a list of given songs into the databse.
+     * 
+     * @param artistDAOProvider
+     *            the artist provider.
+     * @param songDAOProvider
+     *            the song provider.
+     * @param roomDAOProvider
+     *            the room provider.
+     */
+    @Inject
+    public BulkInsertData(final Provider<ArtistDAO> artistDAOProvider,
+            final Provider<SongDAO> songDAOProvider,
+            final Provider<RoomDAO> roomDAOProvider) {
+        this.artistDAOProvider = artistDAOProvider;
+        this.songDAOProvider = songDAOProvider;
+        this.roomDAOProvider = roomDAOProvider;
+        soundCloudExtract = new SoundCloudExtract();
+    }
+
+    /**
+     * Retrieve songs from the API and put the in the database.
+     * 
+     * @throws IOException
+     *             when the file with song ids could not be parsed.
+     * @throws SoundCloudException
+     *             when the SoundCloud API was not called successfully.
+     */
+    @Transactional
+    public void insertData() throws IOException, SoundCloudException {
+        final ArtistDAO artistDAO = artistDAOProvider.get();
+        final SongDAO songDAO = songDAOProvider.get();
+        final VAVector defaultVector = new VAVector(0, 0);
+
+        int[] soundCloudIds = readSoundCloudIds();
+        for (int id : soundCloudIds) {
+            try {
+                SoundCloudTrack track = soundCloudExtract.extract(id);
+                Artist artist = new Artist();
+                artist.setName(track.getUser().getUsername());
+                artistDAO.persist(artist);
+
+                Song song = songToTrack(track, id, artist, defaultVector);
+                songDAO.persist(song);
+            } catch (Exception e) {
+                System.out.println(e.getLocalizedMessage());
+            }
+        }
+    }
+
+    /**
+     * Insert a given amount of random generated rooms into the database. This is used for testing
+     * purposes.
+     * 
+     * @param nRooms
+     *            the amount of rooms to generate.
+     */
+    public void insertRandomRooms(int nRooms) {
+        final RoomDAO roomDAO = roomDAOProvider.get();
+        final SongDAO songDAO = songDAOProvider.get();
+        List<Song> songs = songDAO.listSongs();
+        Random random = new Random();
+        for (int i = 0; i < nRooms; i++) {
+            Room room = new Room();
+            System.out.println(songs.size());
+            room.setCurrentSong(songs.get(random.nextInt(songs.size())));
+            room.setRoomName("ROOM_STUB #" + i);
+            room.setCurrentTime(0);
```

Comment:
I think this line (and the `room.setChatMessages(Collections.<ChatMessage> emptyList());`) are not required.